filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobFileSystemManager.scala

```diff
@@ -96,19 +96,31 @@ case class WSMBlobTokenGenerator(
   container: BlobContainerName,
   endpoint: EndpointURL,
   workspaceId: WorkspaceId,
-  wsmClient: WorkspaceManagerApiClientProvider) extends BlobTokenGenerator {
-
-  def generateAccessToken: Try[AzureSasCredential] = Try {
-    val token = wsmClient.getControlledAzureResourceApi.createAzureStorageContainerSasToken(
-      UUID.fromString(workspaceId.value),
-      UUID.fromString("00001111-2222-3333-aaaa-bbbbccccdddd"),
-      null,
-      null,
-      null,
-      null
-    ).getToken // TODO `null` items may be required, investigate in WX-696
-
-    new AzureSasCredential(token) // TODO Does `signature` actually mean token? save for WX-696
+  containerResourceId: ContainerResourceId,
+  wsmClient: WorkspaceManagerApiClientProvider,
+  defaultB2cToken: Option[String]) extends BlobTokenGenerator {
+
+  def generateAccessToken: Try[AzureSasCredential] = {
+    val controlledAzureResourceApi: Try[ControlledAzureResourceApi] = ((defaultB2cToken) match {
+      case (Some(defaultToken)) => Success(wsmClient.getControlledAzureResourceApi(defaultToken))
+      case (None) => (AzureCredentials(None).getAccessToken.toOption) match {
+        case Some(localAzureCredential) => Success(wsmClient.getControlledAzureResourceApi(localAzureCredential))
+        case None => Failure(new RuntimeException("B2C token not found"))
+      }
+    })
```

Comment:
+1 to `overrideB2CToken`

Not in scope for this PR but we could also consider getting it from an env var or something, passing it in via repo template... should only be used in development environments.

Reply:
You are right default is a little misleading here, I'll try renaming to overrideB2cToken, That is the intent really; in development allow for overriding for a configured token because the Azure environment credentials will not work when developing locally. I can also add a comment